src/main/java/core/basesyntax/dao/figure/FigureDaoImpl.java

```diff
@@ -12,11 +18,35 @@ public FigureDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public T save(T figure) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(figure);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Cannot insert figure: " + figure, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return figure;
     }
 
     @Override
     public List<T> findByColor(String color, Class<T> clazz) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<T> query = criteriaBuilder.createQuery(clazz);
+            Root<T> figureRoot = query.from(clazz);
+            Predicate colorPredicate = criteriaBuilder.like(figureRoot.get("color"), color);
```

Comment:
I think it's better to use `equal` here, not `like`

Reply:
Refactored.